phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexBuildTimestampIT.java

```diff
@@ -0,0 +1,299 @@
+        clientProps.put(QueryServices.STATS_UPDATE_FREQ_MS_ATTRIB, Long.toString(5));
+        clientProps.put(QueryServices.TRANSACTIONS_ENABLED, Boolean.TRUE.toString());
+        clientProps.put(QueryServices.FORCE_ROW_KEY_ORDER_ATTRIB, Boolean.TRUE.toString());
+        setUpTestDriver(new ReadOnlyProps(serverProps.entrySet().iterator()),
+                new ReadOnlyProps(clientProps.entrySet().iterator()));
+    }
+
+    @Parameters(
+            name = "mutable={0},localIndex={1},async={2}")
+    public static Collection<Object[]> data() {
+        List<Object[]> list = Lists.newArrayListWithExpectedSize(8);
+        boolean[] Booleans = new boolean[]{false, true};
+        for (boolean mutable : Booleans) {
+            for (boolean localIndex : Booleans) {
+                for (boolean async : Booleans) {
+                    list.add(new Object[]{mutable, localIndex, async});
+                }
+            }
+        }
+        return list;
+    }
+
+    public static void assertExplainPlan(Connection conn, boolean localIndex, String selectSql,
+                                         String dataTableFullName, String indexTableFullName) throws SQLException {
+        ResultSet rs = conn.createStatement().executeQuery("EXPLAIN " + selectSql);
+        String actualExplainPlan = QueryUtil.getExplainPlan(rs);
+        String expectedExplainPlan;
+        if (localIndex) {
+            expectedExplainPlan = String.format(" RANGE SCAN OVER %s [1,", dataTableFullName);
+        } else {
+            expectedExplainPlan = String.format(" RANGE SCAN OVER %s", indexTableFullName);
+        }
+        assertTrue(actualExplainPlan + "\n expected to contain \n" + expectedExplainPlan,
+                actualExplainPlan.contains(expectedExplainPlan));
+    }
+
+    public static String[] getArgValues(String schemaName,
+                                        String dataTable, String indxTable) {
+        final List<String> args = Lists.newArrayList();
+        if (schemaName != null) {
+            args.add("-s");
+            args.add(schemaName);
+        }
+        args.add("-dt");
+        args.add(dataTable);
+        args.add("-it");
+        args.add(indxTable);
+        args.add("-direct");
+        // Need to run this job in foreground for the test to be deterministic
+        args.add("-runfg");
+        args.add("-op");
+        args.add("/tmp/" + UUID.randomUUID().toString());
+        return args.toArray(new String[0]);
+    }
+
+    public static void runIndexTool(String schemaName,
+                                    String dataTableName, String indexTableName) throws Exception {
+        runIndexTool(schemaName, dataTableName, indexTableName, new String[0]);
+    }
+
+    public static void runIndexTool(String schemaName,
+                                    String dataTableName, String indexTableName, String... additionalArgs) throws Exception {
```

Comment:
Ditto for runIndexTool and getArgs existing in IndexToolIT.
Also, do we need the tenantID parameter test as well? I have a pending PR where we add tenantId as cmd line parameter.

Reply:
I will leverage IndexToolIT methods. This test does not need to use tenantID.